src/cb_admin/pages/AnonUser/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { CommunityBusiness, Visitors, ErrorUtils } from '../../../api';
+import { redirectOnError, renameKeys } from '../../../util';
+import CreateAnonUserForm from './CreateAnonUserForm';
+import PrintOption from './PrintOption';
+
+
+const stage = {
+  ONE: 'ONE',
+  TWO: 'TWO',
+};
+
+export default class AnonUser extends Component {
+
+  state = {
+    name: 'Anonymous ',
+    genders: [],
+    errors: {},
+    stage: stage.ONE,
+  };
+
+  componentDidMount() {
+    CommunityBusiness.update() // used to check cookie permissions
+      .then(() => {
+        const getCb = CommunityBusiness.get({ fields: ['name', 'logoUrl', 'id'] });
+        const getGenders = Visitors.genders();
+
+        return Promise.all([getCb, getGenders]);
+      })
+      .then(([{ data: { result: cbRes } }, { data: { result: gendersRes } }]) =>
+        this.setState({
+          cbOrgName: cbRes.name,
+          cbLogoUrl: cbRes.logoUrl,
+          organisationId: cbRes.id,
+          genders: [{ key: 0, value: '' }].concat(gendersRes.map(renameKeys({ id: 'key', name: 'value' }))),
```

Comment:
This line of code appears around 5 times in the code base, probably time to think if there's a decent way to put that common code.

Looks like the underlying thing we're trying to do is transform an API payload (a list of objects) to a list of object values that can be used as options in a `select` element.

IMO this could live somewhere in `/api/...` or in `/shared/constants.js` since it has something to do with both things. Maybe worth considering if there's any other common/repeated transformations we're doing with API payloads, it could form the basis of an API transform/util module.

Reply:
@eliascodes can you make an issue for this so it can be resolved separately?

Reply:
See #659